### PR TITLE
Opt out Https everywhere in packages.config restore scenarios in all NuGet tools

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes:https://github.com/NuGet/Home/issues/12788

Regression? Last working version:

## Description
When `allowInsecureConnections  `property in `packageSources `section is set to true in NuGet.Config files, suppress warnings for HTTP sources in packages.config restore scenarios(including install and update) in all NuGet tools.

The PRs enable the HTTP warnings in those scenarios are:
https://github.com/NuGet/NuGet.Client/pull/4583
https://github.com/NuGet/NuGet.Client/pull/4633

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 1]
```diff
@@ -1212,6 +1212,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageA);
+            pathContext.Settings.AddSource("http-feed", "http://api.source/index.json", allowInsecureConnections);
+            pathContext.Settings.AddSource("https-feed", "https://api.source/index.json", allowInsecureConnections);
+
+            var net461 = NuGetFramework.Parse("net461");
+            var projectA = new SimpleTestProjectContext(
+                "a",
```
Comment (dtivel) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Pick a project name (e.g.:  `b`) that can't be confused with the package ID (`a`)?  Check other tests too.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 2]
```diff
@@ -1212,6 +1212,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+            using var pathContext = new SimpleTestPathContext();
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageA);
+
+            pathContext.Settings.AddSource("http-feed", "http://api.source/index.json", allowInsecureConnections);
+            pathContext.Settings.AddSource("https-feed", "https://api.source/index.json", allowInsecureConnections);
+
+
```
Comment (dtivel) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Remove extra blank line.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 3]
```diff
@@ -2055,6 +2055,102 @@ public async Task Install_WithPackagesConfigAndHttpSource_Warns()
...
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageA);
+            var packageAPath = Path.Combine(pathContext.PackageSource, packageA.Id, packageA.Version, packageA.PackageName);
+
+            pathContext.Settings.AddSource("http-feed", "http://api.source/index.json", allowInsecureConnections);
+            pathContext.Settings.AddSource("https-feed", "https://api.source/index.json", allowInsecureConnections);
+
+
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Remove extra blank line.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 4]
```diff
@@ -1212,6 +1212,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+            Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""A"" version=""1.0.0"" targetFramework=""net461"" />
+</packages>");
+
+            solution.Projects.Add(projectA);
+            solution.Create(pathContext.SolutionRoot);
+
+            // Act
+            var result = RunRestore(pathContext, _successExitCode);
```
Comment (dtivel) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Use explicit type instead of `var`.  Check other tests too.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 5]
```diff
@@ -2055,6 +2055,102 @@ public async Task Install_WithPackagesConfigAndHttpSource_Warns()
...
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Remove extra blank line.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs [hunk 6]
```diff
@@ -1909,5 +1909,136 @@ public async Task UpdateCommand_WithHttpSource_Warns()
             r.Success.Should().BeTrue(r.AllOutput);
             r.AllOutput.Should().Contain("You are running the 'update' operation with an 'http' source");
         }
+
+        [Theory]
+        [InlineData("true")]
+        [InlineData("TRUE")]
+        public async Task UpdateCommand_WithHttpSourceAndTrueAllowInsecureConnections_NoWarns(string allowInsecureConnections)
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: This and the following test look mostly the same.  In an effort to dedupe code, you could have one test that parameterizes this.  Also consider this feedback elsewhere.

```C#
[Theory]
[InlineData("true", true])
[InlineData("TRUE", true])
[InlineData("false", false])
[InlineData("FALSE", false])
[InlineData("invalid", false])
...
public async Task UpdateCommand_WhenAllowInsecureConnectionsIsSet_WarnsAppropriately(string allowInsecureConnectionsValue, bool allowInsecureConnectionsValue)
{
    ...

    const string warning = "You are running the 'update' operation with an 'http' source";

    if (allowInsecureConnections)
    {
        r.AllOutput.Should().NotContain(warning, because: r.AllOutput);
    }
    else
    {
        r.AllOutput.Should().Contain(warning, because: r.AllOutput);
    }
}
```

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 7]
```diff
@@ -1212,6 +1212,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+
+            // Assert
+            result.Success.Should().BeTrue();
+            Assert.Contains($"Added package 'A.1.0.0' to folder '{projectAPackages}'", result.Output);
+            Assert.Contains("You are running the 'restore' operation with an 'http' source, 'http://api.source/index.json'. Support for 'http' sources will be removed in a future version.", result.Output);
```
Comment (donnie-msft) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: The string in `Warning_HttpServerUsage` uses uppercase `HTTP`. I would think this test would fail, unless Contains is case insensitive.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 8]
```diff
@@ -1212,6 +1212,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageA);
+
+            pathContext.Settings.AddSource("http-feed", "http://api.source/index.json", allowInsecureConnections);
+            pathContext.Settings.AddSource("https-feed", "https://api.source/index.json", allowInsecureConnections);
+
```
Comment (donnie-msft) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs:1232]: nit: whitespace

**Review by donnie-msft — APPROVED** [review]

**Review by dtivel — APPROVED** [review]

**Review by jeffkl — APPROVED** [review]

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 7 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks for reviewing @donnie-msft ! Good catch!
There are 5 `Warning_HttpServerUsage` in our codebase: https://github.com/search?q=repo%3ANuGet%2FNuGet.Client%20name%3D%22Warning_HttpServerUsage%22&type=code
And only one of the 5 is: 
https://github.com/NuGet/NuGet.Client/blob/6e066f0535191e528a724db05758e29982fc909a/src/NuGet.Core/NuGet.PackageManagement/Strings.resx#L433
All other 4 are the same.
The only different one is brought in in PR https://github.com/NuGet/NuGet.Client/pull/4583/files
May I know if there is any reason to have this one different?
Or shall I change it to be the same with the rest 4?

Comment (donnie-msft) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: I suggest using uppercase in all of them (and this is the only one that is lowercase), since it's an acronym. I'm not sure if it's considered a breaking change to edit a warning message? If so, maybe not worth doing at this point?

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 8 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs:1232]: Thanks! Fixed.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 1 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks for reviewing! Fixed.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 2 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks! Fixed.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 3 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Thanks! Fixed.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 4 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks! Fixed.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 5 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Thanks! Fixed.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 7 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks @donnie-msft ! Checked with the team and it's not a breaking change. Fixed.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs (hunk 6 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: Thanks @dtivel ! There are also comments suggesting differently: https://github.com/NuGet/NuGet.Client/pull/5343#discussion_r1284901262
So I think people might have different preferences.

Comment (dtivel) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: I wouldn't view it as a difference in preferences.  The spirit of the comment is to deduplicate code.

I showed one path to deduplication.  Even with the other approach, which is also fine, you can _still_ deduplicate code by extracting reusable code into helper methods.

File: src/NuGet.Core/NuGet.PackageManagement/Strings.resx [hunk 9]
```diff
@@ -430,6 +430,7 @@
     <comment>{0} - package id</comment>
   </data>
   <data name="Warning_HttpServerUsage" xml:space="preserve">
-    <value>You are running the '{0}' operation with an 'http' source, '{1}'. Support for 'http' sources will be removed in a future version.</value>
+    <value>You are running the '{0}' operation with an 'HTTP' source, '{1}'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.</value>
```
Comment (aortiz-msft) [inline src/NuGet.Core/NuGet.PackageManagement/Strings.resx:433]: I don't think this aligns to the new plan. You should check with @JonDouglas  on what the message should be.

**Comment by heng-liu** [comment]
Thanks all for reviewing!
All comments are addressed. Could you please review it again? Thanks!

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 10]
```diff
@@ -1209,7 +1209,90 @@ public async Task Restore_WithHttpSource_Warns()
...
+            result.Success.Should().BeTrue();
+            Assert.Contains($"Added package 'A.1.0.0' to folder '{projectBPackages}'", result.Output);
+            Assert.Contains("You are running the 'restore' operation with an 'HTTP' source, 'http://api.source/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.", result.Output);
+            Assert.DoesNotContain("You are running the 'restore' operation with an 'HTTP' source, 'https://api.source/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.", result.Output);
+        }
+
+        [Theory]
+        [InlineData("true")]
+        [InlineData("TRUE")]
+        public async Task Restore_WithHttpSourceAndTrueAllowInsecureConnections_NoHttpWarns(string allowInsecureConnections)
```
Comment (aortiz-msft) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Do we have negative test cases where using https and setting allowInsecureConnections does NOT warn?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 11]
```diff
@@ -356,11 +356,11 @@ public void InstallCommand_FromPackagesConfigFileWithExcludeVersion()
                 Directory.CreateDirectory(repositoryPath);
 
                 Util.CreateTestPackage("packageA", "1.1.0", repositoryPath);
-                Util.CreateTestPackage("packageB", "2.2.0", repositoryPath);
+                Util.CreateTestPackage("packageA", "2.2.0", repositoryPath);
```
Comment (aortiz-msft) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Are these changes intentional? I didn't see them called out in the PR description. #Closed

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs (hunk 6 above)
Comment (aortiz-msft) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: +1 to what @dtivel suggested.

**Review by aortiz-msft — CHANGES_REQUESTED** [review]

File: src/NuGet.Core/NuGet.PackageManagement/Strings.resx (hunk 9 above)
Comment (heng-liu) [reply src/NuGet.Core/NuGet.PackageManagement/Strings.resx:433]: Thanks @aortiz-msft ! There are 5 Warning_HttpServerUsage in our codebase: https://github.com/search?q=repo%3ANuGet%2FNuGet.Client%20name%3D%22Warning_HttpServerUsage%22&type=code
And only one of the 5 is different. That's why I made this change to make all of them the same.
Hi @JonDouglas , could you help advise if this change is needed? Thanks!

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs (hunk 10 above)
Comment (heng-liu) [reply test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs]: Thanks for reviewing! Yes, we have. Pls refer to line [1229 ](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-optoutHttpWarning-packagesConfigRestore/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs#L1229)and line [1254](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-optoutHttpWarning-packagesConfigRestore/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs#L1254).

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 11 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs:359]: [Comment](https://github.com/NuGet/NuGet.Client/pull/5356#discussion_r1293571230) suggested to make package name and project name different. I first changed package name to B, then changed my mind to change project to B.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs (hunk 6 above)
Comment (donnie-msft) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: The point I was making in the comment you linked is that Setup had many branches, making it hard to understand the path to a test failure. I think the proposal here is to extract setup helper methods, which is OK to consider, and a different topic, really. I did this in [my recent PR](https://github.com/NuGet/NuGet.Client/pull/5295/files#diff-a3a50527eea06e5394c6c390b69b24caa38fe904efe706137acac2ae363f6bbdR406) (eg, `SetupUIServiceWithPackageSearchMetadata`) which had MUCH more complicated Setup, .so it's hard to say it's needed in your PR.

Another viewpoint I don't see in this thread, but which has been expressed in my PR reviews, is that deduping a few lines of code isn't necessarily making it easier for the reader. There's a bit of an art to it. 

I'd say spend just a little effort (<30 minutes?) to see if you can extract methods that saves more than 3-4 lines or so per test. Sometimes you need to return objects and pass them around, and end up with nearly the same setup code. Given, there are 1000+ lines of tests in this file with far more complicated duplicated code, we may want to separately come up with a guideline as a Team, then apply that decision to new tests, and refactor old tests.

File: src/NuGet.Core/NuGet.PackageManagement/Strings.resx (hunk 9 above)
Comment (donnie-msft) [reply src/NuGet.Core/NuGet.PackageManagement/Strings.resx:433]: @aortiz-msft could you point reviewers to what plan you're referring to?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs (hunk 6 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs]: I agree that it would be good to have a guideline as a Team so we will have a consistent standard to follow.
Otherwise, we will have to keep addressing different comments and that is not efficient.
There are many existing tests have the same setup without deduplication: [RestoreCommand_VerifyMinClientVersionV2SourceAsync](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs#L275) and [RestoreCommand_VerifyMinClientVersionV3SourceAsync](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs#L319)
So it's not clear when shall we deduplicate code and when shall we keep duplicate code to make test easy to understand.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 11 above)
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Thanks! I tried to address [this comment](https://github.com/NuGet/NuGet.Client/pull/5356#discussion_r1293571230) but accidently made more changes to other unrelated tests. Fixed.

File: src/NuGet.Core/NuGet.PackageManagement/Strings.resx (hunk 9 above)
Comment (aortiz-msft) [reply src/NuGet.Core/NuGet.PackageManagement/Strings.resx:433]: https://github.com/NuGet/Home/pull/12542/files

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs [hunk 12]
```diff
@@ -1209,7 +1209,62 @@ public async Task Restore_WithHttpSource_Warns()
...
+        }
+
+        [Theory]
+        [InlineData("false", true)]
+        [InlineData("FALSE", true)]
+        [InlineData("invalidString", true)]
+        [InlineData("", true)]
+        [InlineData("true", false)]
+        [InlineData("TRUE", false)]
+        public async Task Restore_WithHttpSourceAndFalseAllowInsecureConnections_WarnsCorrectly(string allowInsecureConnections, bool hasHttpWarning)
```
Comment (aortiz-msft) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs:1222]: Suggestion: isHttpWarningExpected?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs [hunk 13]
```diff
@@ -1907,7 +1907,82 @@ public async Task UpdateCommand_WithHttpSource_Warns()
...
         }
+
+        [Theory]
+        [InlineData("false", true)]
+        [InlineData("FALSE", true)]
+        [InlineData("invalidString", true)]
+        [InlineData("", true)]
+        [InlineData("true", false)]
+        [InlineData("TRUE", false)]
+        public async Task UpdateCommand_WithHttpSourceAndAllowInsecureConnections_WarnsCorrectly(string allowInsecureConnections, bool hasHttpWarning)
```
Comment (aortiz-msft) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs:1920]: Same

File: test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs [hunk 14]
```diff
@@ -1140,7 +1140,75 @@ public async Task MsbuildRestore_PackagesConfigDependency_WithHttpSource_Warns()
...
+        }
+
+        [PlatformTheory(Platform.Windows)]
+        [InlineData("false", true)]
+        [InlineData("FALSE", true)]
+        [InlineData("invalidString", true)]
+        [InlineData("", true)]
+        [InlineData("true", false)]
+        [InlineData("TRUE", false)]
+        public async Task MsbuildRestore_PackagesConfigDependency_WithHttpSourceAndAllowInsecureConnections_WarnsCorrectly(string allowInsecureConnections, bool hasHttpWarning)
```
Comment (aortiz-msft) [inline test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs:1154]: same

**Review by aortiz-msft — APPROVED** [review]
:shipit:

**Review by donnie-msft — APPROVED** [review]
